Tasks/DotNetCoreInstallerV1/versionInstaller.ts

```
@@ -0,0 +1,132 @@
+            tl.debug(tl.loc("CopyingFoldersIntoPath", this.installationPath));
+            var allEnteriesInDir: string[] = fs.readdirSync(extPath).map(name => path.join(extPath, name));
+            var directoriesTobeCopied: string[] = allEnteriesInDir.filter(path => fs.lstatSync(path).isDirectory());
+            directoriesTobeCopied.forEach((directoryPath) => {
+                tl.cp(directoryPath, this.installationPath, "-rf", false);
+            });
+
+            // Copy files
+            try {
+                if (this.isLatestInstalledVersion(version)) {
+                    tl.debug(tl.loc("CopyingFilesIntoPath", this.installationPath));
+                    var filesToBeCopied = allEnteriesInDir.filter(path => !fs.lstatSync(path).isDirectory());
+                    filesToBeCopied.forEach((filePath) => {
+                        tl.cp(filePath, this.installationPath, "-f", false);
+                    });
+                }
+            }
+            catch (ex) {
+                tl.warning(tl.loc("FailedToCopyTopLevelFiles", ex));
+            }
+
+            // Cache tool
+            this.createInstallationCompleteFile(versionInfo);
+
+            console.log(tl.loc("SuccessfullyInstalled", this.packageType, version));
+        }
+        catch (ex) {
+            throw tl.loc("FailedWhileInstallingVersionAtPath", version, this.installationPath, ex);
+        }
+        finally {
+            //todo: Release Lock and stop timer
+        }
+
+        //todo: in case of failure due to: unable to acquire lock or timeout, 3 retries to install.
+    }
+
+    public isVersionInstalled(version: string): boolean {
+        var isInstalled: boolean = false;
+        if (this.packageType == "sdk") {
+            isInstalled = toolLib.isExplicitVersion(version) && fs.existsSync(path.join(this.installationPath, "sdk", version)) && fs.existsSync(path.join(this.installationPath, "sdk", `${version}.complete`));
```

should we throw error if explicit version is not specified